src/votekit/election_types.py

```diff
@@ -139,3 +408,20 @@ def remove_cand(removed_cand: str, ballots: list[Ballot]) -> list[Ballot]:
         update[n].ranking = new_ranking
 
     return update
+
+
+def remove_cand_set(removed_set: Iterable[str], ballots: list[Ballot]) -> list[Ballot]:
```

Comment:
because remove_cand does a deepcopy this method winds up doing a ton of intermediate copies, might be worth altering to either have remove_cand take an iterable[str] or write this without relying on remove_cand and just do your own single deep copy at the start